### ISSUE TITLE
Travis not building

### ISSUE THREAD

**Issue by rettinghaus** [rism-ch/verovio#1383]
On pressing `Show MEI` not the input is shown, but Verovio's page based MEI instead.

**Comment by lpugin** [comment]
I fixed this but travis keeps failing, which means that the develop build is not update. I have no idea what's the issue with travis. I removed all the caches and restarted. Any ideas?

**Comment by rettinghaus** [comment]
No clue. Why do you have Travis running on the gh-pages branch anyway?

**Comment by lpugin** [comment]
> Why do you have Travis running on the gh-pages branch anyway?

It not. The fix was in the toolkit. Maybe @musicEnfanthen has an idea?

**Comment by musicEnfanthen** [comment]
The errors seem to have started with commit https://github.com/rism-ch/verovio/commit/b8e6fa64ea723e6bab31b74d84d7fde7868dc3c0 .

There was an error with the `$jsonOptions` in `toolkit.h`.

The following error was thrown:

```
The command "cmake ../cmake" exited with 0.

18.18s$ make -j 8

Scanning dependencies of target verovio

[  0%] Building CXX object CMakeFiles/verovio.dir/Users/travis/build/rism-ch/verovio/src/accid.cpp.o

[  1%] Building CXX object CMakeFiles/verovio.dir/Users/travis/build/rism-ch/verovio/src/abbr.cpp.o

[  1%] Building CXX object CMakeFiles/verovio.dir/main.cpp.o

<snip>

[  7%] Building CXX object CMakeFiles/verovio.dir/Users/travis/build/rism-ch/verovio/src/beam.cpp.o

/Users/travis/build/rism-ch/verovio/tools/main.cpp:592:45: error: reference to type 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') could not bind to an lvalue of type 'int'

                std::cout << toolkit.GetMEI(page);

                                            ^~~~

/Users/travis/build/rism-ch/verovio/cmake/../include/vrv/toolkit.h:192:43: note: passing argument to parameter 'jsonOptions' here

    std::string GetMEI(const std::string &jsonOptions);

                                          ^

1 error generated.

make[2]: *** [CMakeFiles/verovio.dir/main.cpp.o] Error 1

make[2]: *** Waiting for unfinished jobs....

make[1]: *** [CMakeFiles/verovio.dir/all] Error 2

make: *** [all] Error 2

The command "make -j 8" exited with 2.
cache.2

store build cache
```

After @lpugin fixed the `main.cpp` the make command seems to build fine. But there are still warnings in `view_element.cpp.o` and `view_graph.cpp.o`

````
[ 81%] Building CXX object CMakeFiles/verovio.dir/home/travis/build/rism-ch/verovio/src/view_element.cpp.o

/home/travis/build/rism-ch/verovio/src/view_beam.cpp: In member function ‘void vrv::View::DrawBeamSegment(vrv::DeviceContext*, vrv::BeamSegment*, vrv::BeamDrawingInterface*, vrv::Layer*, vrv::Staff*, vrv::Measure*)’:

/home/travis/build/rism-ch/verovio/src/view_beam.cpp:241:12: warning: variable ‘dy1’ set but not used [-Wunused-but-set-variable]

     double dy1, dy2;

            ^~~

/home/travis/build/rism-ch/verovio/src/view_beam.cpp:241:17: warning: variable ‘dy2’ set but not used [-Wunused-but-set-variable]

     double dy1, dy2;

                 ^~~

[ 82%] Building CXX object CMakeFiles/verovio.dir/home/travis/build/rism-ch/verovio/src/view_graph.cpp.o

At global scope:

cc1plus: warning: unrecognized command line option ‘-Wno-dollar-in-identifier-extension’
```

I am not sure, but maybe this warning causes Travis to fail?

**Comment by musicEnfanthen** [comment]
It actually fails now at the moment when it should update the options list:

```
Update the documentation of the option list

./verovio -? > $OUTPUT_DIRECTORY/_includes/cli.txt
```

**Comment by musicEnfanthen** [comment]
So my first guess is that there is a corrupted option list generated with the make build (see warning at global scope: "unrecognized command line option") and when Travis tries to move it into the OUTPUT_DIRECTORY it fails.

**Comment by musicEnfanthen** [comment]
Ok, it does not seem to be a problem with the option list itself. And also no problem with Travis. 

Tracked it down with GitHub Actions to the exactly same position when the option list is copied to the `cli.txt`. 

`./verovio -?` seems to run fine. But then the line fails with exit code 1.

**Comment by lpugin** [comment]
I think I got it. @th-we  could you double check this ?
https://github.com/rism-ch/verovio/commit/1e1830fde53900b5e60c1c88029870731041534a#diff-36a227146cac398defd63282f9e54084

**Comment by musicEnfanthen** [comment]
Great. 

Maybe it would be better to move the "not recognized" logic into the "default" section instead of the "?" case (help) ?

So something like:
```
case '?':
   display_usage();
   exit(0);
   break;
default: 
   display_usage();
   exit(1);
   break;
```

**Comment by lpugin** [comment]
Finally fixed by reverting #1364
